pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,389 @@
+        """
+        checks the status of the front/back switch, selecting either the front
+        or back mearsurement terminals
+
+        Returns
+        -------
+        front_back : int
+            0 == back terminals
+            1 == front terminals
+
+
+        """
+        status_read=self.status()
+        front_back=status_read[1]>>4&1
+        return front_back
+
+    @property
+    def trigger(self):
+        """
+        Checks the triger settings on the HP3478
+
+        Returns
+        -------
+        trigger_stat : str
+            A string value describing the current status of the trigger
+
+        """
+        status_read=self.status()
+        int_trig=status_read[1]&1
+        ext_trig=status_read[1]>>7&1
+        if int_trig==1 and ext_trig==0:
+            trigger_stat='internal'
+        if int_trig==0 and ext_trig==1:
+            trigger_stat='external'
+        if int_trig==0 and ext_trig==0:
+            trigger_stat='hold'
+        return trigger_stat
+
+    @trigger.setter
+    def trigger(self, trigger_mode):
+        """
+        Offers the possibility to set the trigger mode
+
+        Parameters
+        ----------
+        trigger_mode : str
+            allowed values are:
+                 auto, internal, external, single, hold or fast
+
+        Returns
+        -------
+        None.
+
+        """
+        if trigger_mode in self.TRIGGERS.keys():
+            self.write(self.TRIGGERS[trigger_mode])
+        else:
+            print("fix me")
+
+
+
+    def measure(self,mode='DCV',measurement_range='auto',auto_zero="on",digits=5, trigger='auto'):
+        """
+        returns a measurement result, depeding on the parameters
+
+
+        Parameters
+        ----------
+        mode : str, optional
+            The measurement mode to be used.
+            Valid modes are: 
+                "DCV",
+                "ACV", 
+                "R2W" (2 wire Ohms mode), 
+                "R4W" (4 wire Ohms mode), 
+                "DCI", 
+                "ACI",
+                "Rext" (extended Ohms mode, see manual for more detail)
+            Default is "DCV".
+        measurement_range : str, optional
+            Manaul selection for the measurement range or "auto" for auto-ranging.
+            
+            Default is "auto".
+        auto_zero : int, optional
+            a vlaue of 0 diables autozero, 
+            while a value of 1 enables autozero.
+            
+            Default is 1.
+        digits : int, optional
+            Number of digits selection, allowed values are 3,4,5
+            
+            Default is 5.
+        trigger : str, optional
+            Trigger mode selection, choices are: auto=internal, external, single, hold, fast (see manual for this)
+            
+            Default is 'auto'.
+
+        Returns
+        -------
+        measured_value : float
+            the value mesured based on the mode and other settings.
+            
```

Comment:
I am not familiar with this style of comments, I have seen that in the resto of the code the style used is as follow:
"""[Summary]

:param [ParamName]: [ParamDescription], defaults to [DefaultParamVal]
:type [ParamName]: [ParamType](, optional)
...
:raises [ErrorType]: [ErrorDescription]
...
:return: [ReturnDescription]
:rtype: [ReturnType]
"""
E.g.:
```
        """Formats a record as csv.
        :param record: record to format.
        :type record: dict
        :return: a string
        """
```

Are you sure the style you are using renders well with Sphinx ?

Reply:
You were absolutely right, the way you descibe is the right way to do it.
I was just using the docstring template from spyder, it acutally compiled with sphinx, but there were some warnings.
I have changed the parmeters(&other descriptions).